geps/gep-1619.md

```diff
@@ -159,47 +167,399 @@ Generally, the implementation API programs the dataplane API; however these two
+implementation's objects is hard to directly translate. The idea of a route in Gateway API is often not the same as a
+route in any given implementation.
+
+#### Metaresource Policy Design
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+In order to apply session persistence configuration to both a service and a route, we will implement it as a
+metaresource policy, as outlined in [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment).
+The metaresource is named `SessionPersistence` and is only responsible for configuring session persistence for services.
+It is defined as a [Direct Policy Attachment](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#direct-policy-attachment)
+without defaults or overrides, applied to the service it is targeting.
 
-**Session Cookie Configuration**
+Attaching the `SessionPersistence` metaresource to a service will be a core support level feature, while attaching it to
+a route will be considered extended. This distinction arises because most existing implementations primarily support
+attaching to a service, while attaching to a route is less common and involves greater complexities (see [API Attachment Point: Route vs. Service](#api-attachment-point-route-vs-service)
+for more details).
 
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
+#### API Attachment Point: Route vs. Service
 
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+Session persistence augments both the concepts of routes and services. However, the functionality may vary slightly
+depending on the object to which it is attached. It is important for implementations to understand these differences 
+when implementing the session persistence API. 
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+##### Service
 
-The set-cookies attributes defined by [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+A Kubernetes service is an abstract representation of a set of pods that work together to provide a specific
+functionality. Services perform load balancing, routing traffic from a service endpoint to a pod. Therefore, in its
+simplest form, session persistence, augments the load balancing functionality of the service by consistently directing
+traffic to the same pod. Attaching session persistence configuration to a service has no impact on routing whatsoever.
+
+##### Route
```

Comment:
does it make sense to specify how to handle a case when one session persistence policy targets a Service and another session persistence  targets an HTTPRoute that routes to that Service?

Reply:
Yes, good point, that should be addressed. My gut says that implementation should handle the persistence with the route, as it enables (as I have currently written the in GEP) a "higher level" of functionality. The GEP currently says: If you attach to a route, you get session persistence among all of of your services. So, where routes are effectively a group of services, seems like the higher order configuration should win.

Let me know if this doesn't make sense or you disagree. I'm still a little unsure about whether we should attach to both routes and services and how that might be interpreted by users.

Addressed in next push.

Reply:
Does this one https://gateway-api.sigs.k8s.io/concepts/guidelines/#conflicts help in terms of how to handle conflicts?

Reply:
Thanks for the link. I think that's helpful information. Still tricky to apply to this situation.